drivers/pipes/pipe_common.h

```diff
@@ -135,6 +137,12 @@ struct pipe_dev_s
   struct pollfd *d_fds[CONFIG_DEV_PIPE_NPOLLWAITERS];
 };
 
+enum pipe_common_type_e
+{
+  PIPE_COMMON_PIPE = 0,   /* Pipe */
+  PIPE_COMMON_MKFIFO,     /* Named pipe */
```

Comment:
why the behavior of fifo different from pipe?

Reply:
Because blocking `open` for write-only (with no readers) would fail `pipe` implementation. Please check https://github.com/apache/nuttx/blob/2c36bbc20cda2dbcd75d5992672a10d9bac96568/drivers/pipes/pipe.c#L258.

I made it different to be able to keep the expected behavior of pipes. Does it make sense to you?

Reply:
Why not add O_NONBLOCK temporarily and remove it if caller doesn't really pass O_NONBLOCK, which make the change more simpler.

Reply:
It's a good alternative solution indeed. I've just updated the PR considering your suggestion, please take a look ;)